Transports/com.community.netcode.transport.steamnetworking/Runtime/SteamNetworkingTransport.cs

```diff
@@ -251,83 +244,80 @@ public override NetworkEvent PollEvent(out ulong clientId, out ArraySegment<byte
 #endif
-                                    receiveTime = Time.realtimeSinceStartup;
-                                    return NetworkEvent.Disconnect;
+                                receiveTime = Time.realtimeSinceStartup;
+                                return NetworkEvent.Disconnect;
 
-                                case (byte)InternalChannelType.Connect:
+                            case (byte)InternalChannelType.Connect:
 
-                                    if (isServer)
-                                    {
+                                if (isServer)
+                                {
 #if UNITY_SERVER
                                         SteamGameServerNetworking.SendP2PPacket(remoteId, new byte[] { 0 }, 1, EP2PSend.k_EP2PSendReliable, (int)InternalChannelType.Connect);
 #else
-                                        SteamNetworking.SendP2PPacket(remoteId, new byte[] { 0 }, 1, EP2PSend.k_EP2PSendReliable, (int)InternalChannelType.Connect);
+                                    SteamNetworking.SendP2PPacket(remoteId, new byte[] { 0 }, 1, EP2PSend.k_EP2PSendReliable, (int)InternalChannelType.Connect);
 #endif
-                                    }
-                                    if (connectedUsers.ContainsKey(remoteId.m_SteamID) == false)
-                                    {
-                                        clientId = remoteId.m_SteamID;
-                                        connectedUsers.Add(clientId, new User(remoteId));
-                                        receiveTime = Time.realtimeSinceStartup;
-
-                                        if (!isServer)
-                                        {
-                                            OnConnected();
-                                        }
+                                }
 
+                                if (connectedUsers.ContainsKey(remoteId.m_SteamID) == false)
+                                {
+                                    clientId = remoteId.m_SteamID;
+                                    connectedUsers.Add(clientId, new User(remoteId));
+                                    receiveTime = Time.realtimeSinceStartup;
 
-                                        return NetworkEvent.Connect;
+                                    if (!isServer)
+                                    {
+                                        OnConnected();
                                     }
-                                    break;
 
-                                case (byte)InternalChannelType.Ping:
+                                    return NetworkEvent.Connect;
+                                }
 
-                                    pingPongMessageBuffer[0] = messageBuffer[0];
+                                break;
+
+                            case (byte)InternalChannelType.Ping:
+
+                                pingPongMessageBuffer[0] = messageBuffer[0];
```

Comment:
What is the first byte in that buffer always representing, as I see it used in multiple places?  Can we add a const variable otherwise, without looking at the whole flow it doesn't easily come across what is that all about?

Reply:
Yeah there's a bunch of places like that in the code which could be improved. This PR did just change the control flow of the poll (everything is in the switch now instead of both an if-else and a switch) and not the code itself and I'm frankly not sure why we even need this ping code at all. I suggest that if we have a look we do it in a separate PR.